### PR TITLE
Add keyword arg and Vector2 support to Mask module

### PR DESCRIPTION
**Pull request by Starbuck5**
As I noticed in #2515, the Mask module threw a fit whenever it got a `pygame.Vector2` argument, so my goal was to use the `pg_TwoIntsFromObj` macro to fix this.

I also decided to add support for keyword arguments while I was at it, as per #808.

I added keyword arguments and vector2 testing to `mask_test.py`, not through new cases, but by sprinkling them throughout. This is because `mask_test.py` is already 6400 lines long; it's scary enough already. (Also I was lazy). I also ran `mask_test.py` through the `black` code formatter, which introduced some other changes throughout.

I added two version changed notifications at the top of the Mask docs, to announce these two changes. I thought it would be repetitive to announce the change in every function. (Although the draw docs do this). I also removed several `type` param modifiers in the docs, because now the sequences of two numbers can be input in so many formats I thought it would be unhelpful to list them. The terminology I've used in the TypeError error messages is "a sequence of two numbers."

This also updates the `mask.pyi` type stubs to reflect the change, using the preexisting `_Coordinate = Union[Tuple[float, float], List[float], Vector2]`. I tried to test these new definitions, but I couldn't get mypy to fail at all, even when I gave it bogus like a Vector2 made of strings or a mask offset of a single number.

### CODE REVIEW

**Comment by Starbuck5** [comment]
I think I've fixed everything that was my fault...

When this is ready for merge, feel free to squash merge.

**Comment by illume** [comment]
Cool. Keyword arguments used in the tests really do make the code much more readable!

(I'm a bit afraid of that strange commit at the beginning of this PR.)

> I tried to test these new definitions, but I couldn't get mypy to fail at all, even when I gave it bogus like a Vector2 made of strings or a mask offset of a single number.

This reminds me, I wonder how or if we could start adding some sort of tests for the type definitions?

File: test/mask_test.py [hunk 1]
```diff
@@ -2034,7 +2031,7 @@ def test_convolve__with_output(self):
         o.clear()
         test.clear()
 
-        m.convolve(k, o, (10, 10))
+        m.convolve(othermask=k, outputmask=o, offset=Vector2(10, 10))
```
Comment (illume) [inline test/mask_test.py:2034]: I wonder if these should be `other, output`?

**Review by illume — APPROVED** [review]
I reviewed everything, and it looks good to me :) I'm a bit afraid of that commit at the beginning... (I could squash merge it, and I hope that'd fix it).

There was one naming question, but I think your choice is good.

File: test/mask_test.py (hunk 1 above)
Comment (Starbuck5) [reply test/mask_test.py:2034]: `other` and `output` are good. Having the type name in isn't necessary. I'll fix that up.

**Comment by Starbuck5** [comment]
Thanks.

I think I'm going to continue this in another PR, just like my other change, to get rid of the first commit and to squash all the other commits together.

**Pull request closed without merging**